app/src/adapters/db/postgres_client.py

```diff
@@ -0,0 +1,146 @@
+import logging
+import os
+import urllib.parse
+from typing import Any
+
+import psycopg2
+import sqlalchemy
+import sqlalchemy.pool as pool
+
+from src.adapters.db.client import DBClient
+from src.adapters.db.config import DbConfig, get_db_config
+
+logger = logging.getLogger(__name__)
+
+
+class PostgresDBClient(DBClient):
+    def __init__(self) -> None:
+        self.db_config = get_db_config()
+        super().__init__()
+
+        if self.db_config.check_connection_on_init:
+            self.check_db_connection()
```

Comment:
nit: could we move references to db_config into `_configure_engine`? feels like it belongs there since configure engine has "configure" in the name, and it also feels clean for now to not need to save the db_config as an instance variable, and we also won't need to override the constructor anymore too.

Reply:
I was setting the init up that way because I'm imagining a scenario where we have two clients/DBs that are of the same type.

To make that work, you'd need to make init take in the DB config, but otherwise would work the same. I could make that change, but was just getting it most of the way there at the moment. Thoughts?

Reply:
I adjusted the constructor now